charts/spire/charts/spire-server/templates/tests/test-connection.yaml

```
@@ -11,10 +11,22 @@ spec:
   securityContext:
     {{- toYaml .Values.podSecurityContext | nindent 4 }}
   containers:
-    - name: netcat
+    - name: curl
       image: cgr.dev/chainguard/bash:latest
-      command: ['nc']
-      args: ['-zvw3', '{{ include "spire-server.fullname" . }}', '{{ .Values.service.port }}']
+      command: ['bash']
+      args:
+      - -c
+      - |
+        curl -f -s 'https://{{ include "spire-server.fullname" . }}:{{ .Values.service.port }}'
+        NOCA=$?
+        curl -k -f -s 'https://{{ include "spire-server.fullname" . }}:{{ .Values.service.port }}'
+        IGNORECA=$?
+        echo $NOCA $IGNORECA
+        if [ $NOCA -eq 60 -a $IGNORECA -eq 22 ]; then
+          # We were able to connect to the server but didn't recognize the ca (60) and the page not found (22) because we're not using grpc
+          exit 0
+        fi
+        exit 1
```

```suggestion
        - -c
        - |
          curl -f -s 'https://{{ include "spire-server.fullname" . }}:{{ .Values.service.port }}'
          NOCA=$?
          curl -k -f -s 'https://{{ include "spire-server.fullname" . }}:{{ .Values.service.port }}'
          IGNORECA=$?
          echo $NOCA $IGNORECA
          if [ $NOCA -eq 60 -a $IGNORECA -eq 22 ]; then
            # We were able to connect to the server but didn't recognize the ca (60) and the page not found (22) because we're not using grpc
            exit 0
          fi
          exit 1
```

Nitpick on indentation